sensor/common/registry/registry_store_test.go

```diff
@@ -43,186 +50,264 @@ func alwaysFailCheckTLS(_ context.Context, _ string) (bool, error) {
-	ctx := context.Background()
+func TestRegistryStore_SameNamespace(t *testing.T) {
+	t.Run("SecretsByHost", func(t *testing.T) {
+		testutils.MustUpdateFeature(t, features.SensorPullSecretsByName, false)
+		sameNamespaceSubTest(t)
+	})
+
+	t.Run("SecretsByName", func(t *testing.T) {
+		testutils.MustUpdateFeature(t, features.SensorPullSecretsByName, true)
+		sameNamespaceSubTest(t)
+	})
+}
 
+func sameNamespaceSubTest(t *testing.T) {
 	regStore := NewRegistryStore(alwaysInsecureCheckTLS)
 
-	dce := config.DockerConfigEntry{
-		Username: "username",
-		Password: "password",
+	dce := config.DockerConfigEntry{Username: "username", Password: "password"}
+
+	dc := config.DockerConfig{
+		"image-registry.openshift-image-registry.svc:5000":       dce,
+		"image-registry.openshift-image-registry.svc.local:5000": dce,
+		"172.99.12.11:5000": dce,
 	}
-	require.NoError(t, regStore.UpsertRegistry(ctx, "qa", "image-registry.openshift-image-registry.svc:5000", dce))
-	require.NoError(t, regStore.UpsertRegistry(ctx, "qa", "image-registry.openshift-image-registry.svc.local:5000", dce))
-	require.NoError(t, regStore.UpsertRegistry(ctx, "qa", "172.99.12.11:5000", dce))
+
+	regStore.UpsertSecret("qa", fakeSecretName, dc, noServiceAcctName)
 
 	img := &storage.ImageName{
 		Registry: "image-registry.openshift-image-registry.svc:5000",
 		Remote:   "qa/nginx",
 		Tag:      "nginx:1.18.0",
 		FullName: "image-registry.openshift-image-registry.svc:5000/qa/nginx:1.18.0",
 	}
-	reg, err := regStore.GetRegistryForImageInNamespace(img, "qa")
+	regs, err := regStore.GetPullSecretRegistries(img, "qa", nil)
 	require.NoError(t, err)
-	assert.Equal(t, "image-registry.openshift-image-registry.svc:5000", reg.Config(ctx).RegistryHostname)
+	require.Len(t, regs, 1)
+	assert.Equal(t, img.GetRegistry(), regs[0].Config(bgCtx).RegistryHostname)
 
 	img = &storage.ImageName{
 		Registry: "image-registry.openshift-image-registry.svc.local:5000",
 		Remote:   "qa/nginx",
 		Tag:      "nginx:1.18.0",
 		FullName: "image-registry.openshift-image-registry.svc.local:5000/qa/nginx:1.18.0",
 	}
-	reg, err = regStore.GetRegistryForImageInNamespace(img, "qa")
+
+	regs, err = regStore.GetPullSecretRegistries(img, "qa", nil)
 	require.NoError(t, err)
-	assert.Equal(t, "image-registry.openshift-image-registry.svc.local:5000", reg.Config(ctx).RegistryHostname)
+	require.Len(t, regs, 1)
+	assert.Equal(t, img.GetRegistry(), regs[0].Config(bgCtx).RegistryHostname)
 
 	img = &storage.ImageName{
 		Registry: "172.99.12.11:5000",
 		Remote:   "qa/nginx",
 		Tag:      "nginx:1.18.0",
 		FullName: "172.99.12.11:5000/qa/nginx:1.18.0",
 	}
-	reg, err = regStore.GetRegistryForImageInNamespace(img, "qa")
+	regs, err = regStore.GetPullSecretRegistries(img, "qa", nil)
 	require.NoError(t, err)
-	assert.Equal(t, "172.99.12.11:5000", reg.Config(ctx).RegistryHostname)
+	require.Len(t, regs, 1)
+	assert.Equal(t, img.GetRegistry(), regs[0].Config(bgCtx).RegistryHostname)
 }
```

Comment:
can you add a test for a registry which should not exist?

Reply:
The test already existed (in the subsequent test), there was some redundancy in the tests - I've cleaned this up in the latest commit, please take a look.